### ISSUE TITLE
bug(Node): ESLint devDirs does not actually allow anything but dirs, despite the description

### ISSUE THREAD

**Issue by dkershner6** [projen/projen#3210]
> Files or glob patterns or directories with source files that include tests and build tools
>  These sources are linted but may also import packages from `devDependencies`.

Providing `src/**/*.test.ts` outputs `**/src/**/*.test.ts/**`.  This makes it impossible to pick out files mixed into the same directory as standard src files.

https://github.com/projen/projen/blob/0f2c3605698420b632b132e21792ae538e141601/src/javascript/eslint.ts#L201

**Comment by mrgrain** [comment]
Yeah I think we should deprecate this property in favor of `devPatterns`. Looks like I took a shortcut here a while ago 🙊

**Comment by dkershner6** [comment]
> Yeah I think we should deprecate this property in favor of `devPatterns`. Looks like I took a shortcut here a while ago 🙊

🙈 

Maybe `devDepAllowPatterns`?  In the context of ESLint, `devPatterns` could be too many things, IMO.

There is also a `dirs`, but to be honest, I haven't looked at what it does.

**Comment by mrgrain** [comment]
> Maybe `devDepAllowPatterns`? In the context of ESLint, `devPatterns` could be too many things, IMO.
>
> There is also a `dirs`, but to be honest, I haven't looked at what it does.

The eslint component (attempts to) makes the following abstraction:

- `dirs` - Give me a list of matching patterns that should be linted as production files. 
- `devDirs` - These aren't production files, so please use less strict rules for these.

Granted, it's not doing much else with `devDirs` at the moment, but that's the general idea and it's the approach projen wants to take. A user shouldn't have to know about this allow dev dependency eslint rule.

**Comment by dkershner6** [comment]
It also overwrites the array, rather than adds.  I have seen both behaviors with arrays, but this one feels closer to `tsconfig` `includes`, which is additive.

**Comment by mrgrain** [comment]
> It also overwrites the array, rather than adds. 

How do you mean?

**Comment by dkershner6** [comment]
> How do you mean?

```
{
devDirs: ["momo"]
}
```
=> `["**/momo/**"]`

`project.eslint.allowDevDeps("**/momo/**")`
=> `["**/test/**","**/build-tools/**","**/momo/**"]`
